static/js/prompt.js

```diff
@@ -347,23 +345,50 @@ var app = new Vue({
         },
 
         toggleSort: function(tab) {
+            let url = new URL(window.location.href)
             if (tab === 'time' && this.sortMode === 'path') {
-                if (this.run_id) {
-                    window.location.replace(window.location.href.split('?')[0] + "?run_id=" + this.run_id);
-                } else {
-                    window.location.replace(window.location.href.split('?')[0]);
-                }
+                url.searchParams.set('sort', 'time')
             } else if (tab === 'path' && this.sortMode === 'time') {
-                if (this.run_id) {
-                    window.location.replace(window.location.href.split('?')[0] + "?run_id=" + this.run_id + "&sort=path");
-                } else {
-                    window.location.replace(window.location.href.split('?')[0] + "?sort=path");
-                }
+                url.searchParams.set('sort', 'path')
+            }
+
+            window.location.replace(url)
+        },
+
+        toggleTimeFilter: function(f) {
+            if (f != this.timeFilter) {
+                let url = new URL(window.location.href)
+                url.searchParams.set('time_filter', f)
+                window.location.replace(url)
             }
         },
 
         runReplay: function(event) {
             console.log(event)
+        },
+
+        filterByTime: function() {
+
+            let now = Date.now()
+
+            this.runs.forEach(el => {
+                let date = Date.parse(el.end_time)
+                console.log((now - date) / (1000 * 60 * 60 * 24))
+            });
+
+            if (!['1', '7', '30', '100'].includes(this.timeFilter)) return;
```

Comment:
Doing the filtering on the front end is fine here I think, but eventually we want to move it to the backend